app/(dashboard)/createInvoice/page.jsx

```diff
@@ -0,0 +1,1013 @@
+  ]);
+
+  const handleAddItem = () => {
+    setItems([
+      ...items,
+      { itemDescription: "", quantity: "", price: "", amount: "" },
+    ]);
+  };
+
+  const handleItemsInputChange = (index, event) => {
+    const { id, value } = event.target;
+    const newItems = [...items];
+    newItems[index][id] = value;
+
+    if (errors.items) {
+      const itemErrors = errors.items.map((itemError, idx) =>
+        idx === index ? { ...itemError, [id]: false } : itemError
+      );
+      setErrors((prev) => ({ ...prev, items: itemErrors }));
+    }
+
+    // Calculate the amount if quantity or price changes
+    if (id === "quantity" || id === "price") {
+      const quantity = newItems[index].quantity
+        ? parseFloat(newItems[index].quantity)
+        : 0;
+      const price = newItems[index].price
+        ? parseFloat(newItems[index].price)
+        : 0;
+      newItems[index].amount = quantity * price;
+    }
+
+    setItems(newItems);
+  };
+
+  const handleDiscountChange = (e) => {
+    const value = e.target.value;
+    const calculatedDiscount = (
+      parseFloat(value) *
+      (subTotal / 100)
+    ).toString();
+    setDiscount(calculatedDiscount);
+
+    if (errors.discount) setErrors((prev) => ({ ...prev, discount: false }));
+  };
+
+  const handleTaxChange = (e) => {
+    const value = e.target.value;
+    const calculatedTax = (parseFloat(value) * (subTotal / 100)).toString();
+    setTax(calculatedTax);
+
+    if (errors.tax) setErrors((prev) => ({ ...prev, tax: false }));
+  };
+
+  useEffect(() => {
+    const total = items.reduce((sum, item) => sum + Number(item.amount), 0);
+    setSubTotal(total);
+  }, [items]);
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    const newErrors = {};
+    if (!name) newErrors.name = true;
+    if (!email) newErrors.email = true;
+    if (!customerName) newErrors.customerName = true;
+    if (!billingAddress) newErrors.billingAddress = true;
+    if (!phoneNumber) newErrors.phoneNumber = true;
+    if (!customerEmail) newErrors.customerEmail = true;
+    if (!invoiceTitle) newErrors.invoiceTitle = true;
+    if (!paymentCurrency) newErrors.paymentCurrency = true;
+    if (!accountName) newErrors.accountName = true;
+    if (!accountNumber) newErrors.accountNumber = true;
+    if (!bankName) newErrors.bankName = true;
+    if (!issueDate) newErrors.issueDate = true;
+    if (!dueDate) newErrors.dueDate = true;
+    if (!discount) newErrors.discount = true;
+    if (!tax) newErrors.tax = true;
+
+    const itemErrors = items.map((item) => ({
+      itemDescription: !item.itemDescription,
+      quantity: !item.quantity,
+      price: !item.price,
+    }));
+
+    if (itemErrors.some((error) => Object.values(error).includes(true))) {
+      newErrors.items = itemErrors;
+    }
+
+    if (Object.keys(newErrors).length > 0) {
+      setErrors(newErrors);
+      return;
+    }
+
+    const payload = {
```

Comment:
I had added brandLogo here and i can't fine it. My changes are not in this file. Your are not doing something well here